core/shared/src/main/scala/org/http4s/internal/CurlConverter.scala

```diff
@@ -60,4 +60,22 @@ private[http4s] object CurlConverter {
 
     s"curl$params"
   }
+
+  def requestToCurlWithBody[F[_]: MonadThrow](
+      request: Request[F],
+      redactHeadersWhen: CIString => Boolean,
+  )(implicit compiler: fs2.Compiler[F, F], defaultCharset: Charset): F[(String, Request[F])] =
```

Comment:
Usually, we ask for the `Concurrent` instance, and fs2 internally derives needed underlying instances. I think it's an acceptable approach to use here too.

Reply:
What is the purpose to return `Request[F]`? Isn't this method for debugging only?

Reply:
I didn't want to make the choice for the caller between `Sync` and `Concurrent`, but I'm comfortable just going with `Concurrent`.

Returning a `Request[F]` is because this consumes the body, which might cause trouble if you want to log the request and also do something with it. To avoid that we cache the body and re-stream it in a new `Request[F]`.

Reply:
See discussion in:
- https://github.com/typelevel/fs2/issues/2618
- https://github.com/http4s/http4s/pull/5206

Reply:
Done